### DISCUSSION TITLE
Question about the scheduling.

### DISCUSSION THREAD

**Discussion by He-Pin** [lampepfl/gears#58] [General]
Hi I see the current implementation is using a VT to do the `schedule(...)` and which will use the `ScheduledExecutorService UNPARKER` after all, but IIRC, that will cause problem for large timeout/scheduling set.

And also, today, cats-effect/Netty is integrating the scheduling with WorkThread or EventLoop instead of another `ScheduledExecutorService` for performance ,but we d a rescheduling in gears, is that by design.

**Comment by natsukagami** [comment]
I did not know about `ScheduledExecutorService`'s scaling problem, please enlighten me!

On the other hand, gears should be fairly flexible when it comes to providing a scheduler (see `AsyncSupport`). It was designed with the goal of supporting different schedulers on Scala Native in mind, therefore by default it should support multiple scheduler implementations on JVM as well. 

The one provided by VTScheduler is simply the default implementation (enabled if you import `gears.async.default.given`).
